src/System.Private.CoreLib/src/System/Diagnostics/Eventing/ManagedRuntimeEventSource.cs

```diff
@@ -0,0 +1,115 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+
+using System.Runtime.InteropServices;
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+
+using System.Runtime.InteropServices.ComTypes;
+using System.StubHelpers;
+
+
+namespace System.Diagnostics.Tracing
+{
+#if FEATURE_PERFTRACING
+    /// <summary>
+    /// ManagedRuntimeEventSource is an EventSource that represents events emitted by the managed runtime.
+    /// </summary>
+    [EventSource(Name = "System.Runtime")]
+    internal sealed class ManagedRuntimeEventSource : EventSource
```

Comment:
Do we need another EventSource rather than just using e.g. FrameworkEventSource?

Reply:
There are a couple of factors that argue that we should create a new EventSource for the performance counters.

1. While it is possible to have both monitoring and high volume events in the same EventSource and control them with keywords, the fact of the matter is that users are very poor at using keywords effectively (they tend to turn them all on or all off).   That argues that if you have EventSources that are used for quite different customers, having differnet eventSources has value.   This is especially true if some customers just need low volume events, and there are high volume events that they could easly accidentally turn on.     FrameworkEventSource does have high volume events (the threadpool and networking events), and so there is value in having a separate one.   
2. FrameworkEventSource was one of the first EventSources, and did not follow conventions that were developed later.   Most importantly its GUID was not formed by a hash of its name.  This means that unlike almost every other eventSoruce (TPL might have the same issue), you can't use its name to turn it on (you need to know its speical GUID.   It also did not follow naming conventions (it is OK to have 'EventSource in the classname, but we don't FrameworkEventSource put 'EventSource' in the name used to control it (which we discourage).   Thus having another EventSource allows us to fix these issues.  
3. Having another eventSource allows use to make that one the 'friendly' one, including only those events that probably make sense to all users.   FrameworkEventSource can thus be used as the 'advanced' eventSource, but ManagedEventSource for simpler things we wish to promote more agressively.   

In the end it is a judgement call.   While I agree having many EventSources is bad, having 2 rather than 1 is not a huge deal.    I think the issues above argue for 2.